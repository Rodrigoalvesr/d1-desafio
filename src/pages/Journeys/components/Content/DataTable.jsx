import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import Table from '@material-ui/core/Table';
import TableBody from '@material-ui/core/TableBody';
import TableCell from '@material-ui/core/TableCell';
import TableContainer from '@material-ui/core/TableContainer';
import TableHead from '@material-ui/core/TableHead';
import TableRow from '@material-ui/core/TableRow';
import Skeleton from '@material-ui/lab/Skeleton';
import Status from '@Components/Status';

const useStyles = makeStyles({
  table: {
    width: 900,
  },
  th: {
    fontWeight: 'bold',
    width: '40%',
  },
});

function DataTable({ data, isLoading }) {
  const classes = useStyles();

  return (
    <TableContainer>
      <Table className={classes.table}>
        <TableHead>
          <TableRow>
            <TableCell>Nome</TableCell>
            <TableCell>Destinat√°rio</TableCell>
            <TableCell>Sucesso</TableCell>
            <TableCell>Status</TableCell>
          </TableRow>
        </TableHead>
        <TableBody>
          {isLoading
            ? new Array(4).fill(true).map((item, index) => (
                <TableRow key={index}>
                  <TableCell component="th" scope="row" className={classes.th}>
                    <Skeleton width={120} />
                  </TableCell>
                  <TableCell>
                    <Skeleton width={70} />
                  </TableCell>
                  <TableCell>
                    <Skeleton width={30} />
                  </TableCell>
                  <TableCell>
                    <Skeleton width={80} />
                  </TableCell>
                </TableRow>
              ))
            : data.map((item) => (
                <TableRow key={item.id} className={classes.row}>
                  <TableCell component="th" scope="row" className={classes.th}>
                    {item.name}
                  </TableCell>
                  <TableCell>{item.recipients}</TableCell>
                  <TableCell>{item.success}</TableCell>
                  <TableCell>
                    <Status id={item.status} />
                  </TableCell>
                </TableRow>
              ))}
        </TableBody>
      </Table>
    </TableContainer>
  );
}

export default DataTable;
