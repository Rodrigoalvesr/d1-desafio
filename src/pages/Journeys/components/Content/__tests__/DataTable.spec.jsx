import React from 'react';
import renderer from 'react-test-renderer';
import DataTable from '../DataTable';

describe('DataTable', () => {
  it('Renders correctly', () => {
    const mockValues = [
      {
        name: 'Crédito pré-aprovado',
        status: 5,
        recipients: '25.000',
        success: '23%',
        id: '5f2311e88e6b8b24e047de91',
      },
    ];
    const component = renderer.create(<DataTable data={mockValues} />).toJSON();
    expect(component).toMatchSnapshot();
  });

  it('Renders correctly loading', () => {
    const mockValues = [];
    const component = renderer
      .create(<DataTable data={mockValues} isLoading />)
      .toJSON();
    expect(component).toMatchSnapshot();
  });
});
