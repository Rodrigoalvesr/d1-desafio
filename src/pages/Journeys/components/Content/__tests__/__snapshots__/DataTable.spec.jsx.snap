// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataTable Renders correctly 1`] = `
<div
  className="MuiTableContainer-root"
>
  <table
    className="MuiTable-root makeStyles-table-1"
    role={null}
  >
    <thead
      className="MuiTableHead-root"
      role={null}
    >
      <tr
        className="MuiTableRow-root MuiTableRow-head"
        role={null}
      >
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head"
          scope="col"
        >
          Nome
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head"
          scope="col"
        >
          Destinatário
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head"
          scope="col"
        >
          Sucesso
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head"
          scope="col"
        >
          Status
        </th>
      </tr>
    </thead>
    <tbody
      className="MuiTableBody-root"
      role={null}
    >
      <tr
        className="MuiTableRow-root"
        role={null}
      >
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body makeStyles-th-2"
          role="cell"
          scope="row"
        >
          Crédito pré-aprovado
        </th>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body"
        >
          25.000
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body"
        >
          23%
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body"
        >
          <div
            className="makeStyles-root-3"
          >
            <svg
              className="makeStyles-root-6 makeStyles-root-7 makeStyles-icon-4"
            >
              check.svg
            </svg>
            <p
              className="MuiTypography-root makeStyles-name-5 MuiTypography-body1"
            >
              Concluida
            </p>
          </div>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`DataTable Renders correctly loading 1`] = `
<div
  className="MuiTableContainer-root"
>
  <table
    className="MuiTable-root makeStyles-table-1"
    role={null}
  >
    <thead
      className="MuiTableHead-root"
      role={null}
    >
      <tr
        className="MuiTableRow-root MuiTableRow-head"
        role={null}
      >
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head"
          scope="col"
        >
          Nome
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head"
          scope="col"
        >
          Destinatário
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head"
          scope="col"
        >
          Sucesso
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head"
          scope="col"
        >
          Status
        </th>
      </tr>
    </thead>
    <tbody
      className="MuiTableBody-root"
      role={null}
    >
      <tr
        className="MuiTableRow-root"
        role={null}
      >
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body makeStyles-th-2"
          role="cell"
          scope="row"
        >
          <span
            className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
            style={
              Object {
                "height": undefined,
                "width": 120,
              }
            }
          />
        </th>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body"
        >
          <span
            className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
            style={
              Object {
                "height": undefined,
                "width": 70,
              }
            }
          />
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body"
        >
          <span
            className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
            style={
              Object {
                "height": undefined,
                "width": 30,
              }
            }
          />
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body"
        >
          <span
            className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
            style={
              Object {
                "height": undefined,
                "width": 80,
              }
            }
          />
        </td>
      </tr>
      <tr
        className="MuiTableRow-root"
        role={null}
      >
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body makeStyles-th-2"
          role="cell"
          scope="row"
        >
          <span
            className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
            style={
              Object {
                "height": undefined,
                "width": 120,
              }
            }
          />
        </th>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body"
        >
          <span
            className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
            style={
              Object {
                "height": undefined,
                "width": 70,
              }
            }
          />
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body"
        >
          <span
            className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
            style={
              Object {
                "height": undefined,
                "width": 30,
              }
            }
          />
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body"
        >
          <span
            className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
            style={
              Object {
                "height": undefined,
                "width": 80,
              }
            }
          />
        </td>
      </tr>
      <tr
        className="MuiTableRow-root"
        role={null}
      >
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body makeStyles-th-2"
          role="cell"
          scope="row"
        >
          <span
            className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
            style={
              Object {
                "height": undefined,
                "width": 120,
              }
            }
          />
        </th>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body"
        >
          <span
            className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
            style={
              Object {
                "height": undefined,
                "width": 70,
              }
            }
          />
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body"
        >
          <span
            className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
            style={
              Object {
                "height": undefined,
                "width": 30,
              }
            }
          />
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body"
        >
          <span
            className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
            style={
              Object {
                "height": undefined,
                "width": 80,
              }
            }
          />
        </td>
      </tr>
      <tr
        className="MuiTableRow-root"
        role={null}
      >
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body makeStyles-th-2"
          role="cell"
          scope="row"
        >
          <span
            className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
            style={
              Object {
                "height": undefined,
                "width": 120,
              }
            }
          />
        </th>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body"
        >
          <span
            className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
            style={
              Object {
                "height": undefined,
                "width": 70,
              }
            }
          />
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body"
        >
          <span
            className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
            style={
              Object {
                "height": undefined,
                "width": 30,
              }
            }
          />
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body"
        >
          <span
            className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
            style={
              Object {
                "height": undefined,
                "width": 80,
              }
            }
          />
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
